### DISCUSSION TITLE
How do I remove the default effects of bevy_mod_picking?

### DISCUSSION THREAD

**Discussion by dolto** [bevyengine/bevy#10257] [Q&A] [answered]
Hi, I'm trying to make a hex 3d map using a hexagonal library.

As I looked at the example, I joined this with the mesh that I have, and I ended up using another library, bevy_mod_picking (ver.0.15), to write a click-related method.

The following events allow the mouse to enter and exit the entity.

```rust
commands.spawn((
                    PbrBundle {
                    transform: Transform::from_xyz(pos.x, 0., pos.y)
                        .with_scale(Vec3::splat(0.9)),
                    mesh: mesh_handle.clone(),
                    material: mat,
                    ..default()
                    },
                    Honeycomb,
                    PickableBundle::default(),
                    RaycastPickTarget::default(),
                    On::<Pointer<Over>>::run(on_over),
                    On::<Pointer<Out>>::run(on_out)
))

fn on_over(
    mut commands: Commands,
    event: Listener<Pointer<Over>>,
    grid: Res<Map>,
){
    commands.entity(event.target).insert(grid.path_mat.clone());
}

fn on_out(
    mut commands: Commands,
    event: Listener<Pointer<Out>>,
    grid: Res<Map>,
){
    if grid.blocked_coords.get(&grid.entities_forentity[&event.target]).is_some(){
        commands.entity(event.target).insert(grid.blocked_mat.clone());
    }
    else if grid.path_entities.get(&event.target).is_some(){
        commands.entity(event.target).insert(grid.path_mat.clone());
    }
    else{
        commands.entity(event.target).insert(grid.default_mat.clone());
    }
}
``` 

The code works fine, but I want to eliminate the following basic effects.

![image](https://github.com/bevyengine/bevy/assets/43398206/83ddef56-673e-4077-991b-c4c8944c9103)

As the document says, (if I understand correctly) to lighten up effects like debug. I modified the TOML file as follows.

```rust
bevy_mod_picking = {version = "0.15.0", feature = ["backend_raycast", "backend_bevy_ui"]}
``` 

However, this issue has not been resolved.
What should I do?

**Comment by keis** [comment]
It's a default feature so you would also have to add ` default-features = false`  to remove the defaults and then add what you need.

alternatively you can disable it an runtime like this
```
app.add_plugins(DefaultPickingPlugins.build().disable::<DebugPickingPlugin>())
```

**Reply by dolto** [reply]
I wanted to disable not only debug functions, but also other features that don't need to be compiled, so I tried the following.

```
bevy_mod_picking = {version = "0.15.0", default-features = false, feature = ["backend_raycast", "backend_bevy_ui"]}
``` 

However, this will cause an error in the following code.

```rust
use bevy_mod_picking::prelude::RaycastPickCamera;
//and...
use bevy_mod_picking::prelude::RaycastPickTarget;
``` 

In the case of RaycastPickCamera and RaycastPickTarget, I thought it would be a feature included in "backend_raycast" if we infer it from the name.

To test, I wrote the following in Toml file, but the problem was not solved.
```
bevy_mod_picking = {version = "0.15.0", default-features = false, feature = ["backend_raycast", "backend_shader","backend_bevy_ui", "highlight", "selection"]}
``` 
What can I do?

**Reply by komadori** [reply, marked as answer]
The key in the TOML file is called ```features```, but you appear to have written ```feature``` without the "s".

See https://doc.rust-lang.org/cargo/reference/features.html#dependency-features

**Reply by dolto** [reply]
I actually screamed when I saw this answer.
's'... I can't believe this was the problem... And I'm ashamed that I didn't notice that.

Thank you for your reply. TT
